packages/core/src/indexing/local-index-api.ts

```diff
@@ -49,15 +49,22 @@ export class LocalIndexApi implements IndexApi {
    *
    * We assume that a state store always contains StreamState for an indexed stream.
    */
-  async queryIndex(query: BaseQuery & Pagination): Promise<Page<StreamState>> {
+  async queryIndex(query: BaseQuery & Pagination): Promise<Page<StreamState | null>> {
     if (this.databaseIndexApi) {
       const page = await this.databaseIndexApi.page(query)
       const edges = await Promise.all(
         // For database queries we bypass the stream cache and repository loading queue
         page.edges.map(async (edge) => {
+          let node: StreamState = null
+          try {
+            node = await this.repository.streamState(edge.node)
```

Comment:
It does not throw on empty state. It returns `undefined`. So, instead of catching an error, here it should be comparison to `undefined`.

Even if the code here is changed, the change will not completely eliminate the error we see on the API.

LocalIndexAPI#queryIndex result is fed to https://github.com/ceramicnetwork/js-ceramic/blob/develop/packages/cli/src/ceramic-daemon.ts#L484 which throws the observed error. So, we should fix that part in CeramicDaemon.

TBH, that should have been caught by the compiler. I expected it to be caught really. `e.node` there is nullable, it is why it throws.